data_aggregation.py

```diff
@@ -1,245 +1,82 @@
     
     * All lists must be same length
     '''
-    # Convert all input lists into numpy arrays
-    sd_list = np.array(sd_list)
-    mean_list = np.array(mean_list)
-    pop_list = np.array(pop_list)
-    weight_list = np.array(weight_list)
     # Check for equal sized lists
-    if not len(sd_list) == len(mean_list)==len(pop_list)==len(weight_list):
+    if not len(sd_list) == len(mean_list)==len(pop_list):
         print('All input lists to the function -- combine_standard_deviations -- must be of the same length.')
-    # Proceed with the program
-    # Compute the weighted mean of the populations
-    pop_mean = combine_means(mean_list, pop_list, weight_list)
+    # Compute the weighted mean by populations
+    pop_mean = np.sum(mean_list*pop_list)/(np.sum(pop_list))
     # Compute the deviance
-    deviance = np.sum((pop_list)*(weight_list)*(sd_list**2) + (pop_list*weight_list)*(mean_list - pop_mean)**2)
+    deviance = np.sum((pop_list)*(sd_list**2) + (pop_list)*(mean_list - pop_mean)**2)
     # Compute the standard deviation
-    sd = np.sqrt(deviance/(np.sum(pop_list*weight_list)))
-    return pop_mean,sd
-
+    sd = np.sqrt(deviance/(np.sum(pop_list)))
+    return sd
 
 def aggregate_data(df):
-
-    # Drop rows where the NaN value exists from the dataframe
-    df.dropna(inplace=True)
-    # We can't handle zero responses, it doesnt make sense
-    df['Responses'].replace(to_replace=0, value=1, inplace=True)
-
-    # Initialize the aggregated dataframe by copying the base data frame
-    ag_df = df.copy()
-
-    # Drop the unnecessary columns
-    columns_used = ['Responses','Section Title', 'College Code','Subject Code', 'Term Code','Course Number', 'Instructor ID', 'Instructor First Name', 'Instructor Last Name']
-    columns_to_drop = list(df.columns)
-    for i in columns_used:
-        try:
-            columns_to_drop.remove(i)
-        except:
-            raise Exception('In data_aggregation.py, the input df does not contain the column '+i+', which is needed for the aggregation')
-    ag_df.drop(columns_to_drop, axis=1, inplace = True, errors='ignore')     
-        
-    # Add in the columns to be filled with the aggregated values
-    ag_df.insert(3,'Avg Department Rating', 0.0)
-    ag_df.insert(4,'SD Department Rating', 0.0)
-    ag_df.insert(7,'Avg Course Rating', 0.0)
-    ag_df.insert(8,'SD Course Rating', 0.0)
-    ag_df.insert(9,'Course Rank in Department in Semester', 0)
-    ag_df.insert(12, 'Avg Instructor Rating In Section', 0.0)
-    ag_df.insert(13, 'SD Instructor Rating In Section', 0.0)
-    ag_df.insert(15, 'Course Enrollment', 0)
-
-    # Rename the necessary columns
-    ag_df.rename(columns = {'Section Title':'Course Title', 'Responses':'Instructor Enrollment'}, inplace= True)
+    """
+    Aggregates a pandas dataframe of student reviews data. See the First, Second, and Third Operations below for more descriptions of functions.
+    Note the similar form across operations.
+    :Inputs:
```

Comment:
Idc what docstrings we use really, but might be a good idea to just do PEP. I think they endorse reST. 

Here:
```python3
"""
This is a reST style.

:param param1: this is a first param
:param param2: this is a second param
:returns: this is a description of what is returned
:raises keyError: raises an exception
"""
```

Reply:
Appreciate the PEP stuff; Don't have much experience following standards in developing software but its v helpful.